src/main/fc/rc.c

```diff
@@ -489,141 +489,197 @@ static FAST_CODE void processRcSmoothingFilter(void)
 }
 #endif // USE_RC_SMOOTHING_FILTER
 
-NOINLINE void initAveraging(uint16_t feedforwardAveraging)
+// should this be inside ifdef USE_FEEDFORWARD ??
+NOINLINE void initFeedforwardFilters(uint16_t feedforwardAveraging)
 {
-    for (int i = 0; i < XYZ_AXIS_COUNT; i++) {
-        laggedMovingAverageInit(&feedforwardDeltaAvg[i].filter, feedforwardAveraging + 1, (float *)&feedforwardDeltaAvg[i].buf[0]);
+    if (feedforwardAveraging) {
+        for (int i = 0; i < XYZ_AXIS_COUNT; i++) {
+            laggedMovingAverageInit(&feedforwardDeltaAvg[i].filter, feedforwardAveraging + 1, (float *)&feedforwardDeltaAvg[i].buf[0]);
+        }
     }
+    pt1FilterInit(&feedforwardYawHoldLpf, 0.0f);
 }
 
+#ifdef USE_FEEDFORWARD
 FAST_CODE_NOINLINE void calculateFeedforward(const pidRuntime_t *pid, int axis)
 {
     const float rxInterval = currentRxIntervalUs * 1e-6f; // seconds
-    float rxRate = currentRxRateHz;
-    static float prevRxInterval;
-
+    float rxRate = currentRxRateHz;                 // 1e6f / currentRxIntervalUs;
     static float prevRcCommand[3];
     static float prevRcCommandDeltaAbs[3];          // for duplicate interpolation
-    static float prevSetpoint[3];                   // equals raw unless interpolated 
+    static float prevSetpoint[3];                   // equals raw unless extrapolated forward 
     static float prevSetpointSpeed[3];
-    static float prevAcceleration[3];               // for duplicate interpolation
+    static float prevSetpointSpeedDelta[3];               // for duplicate extrapolation
     static bool prevDuplicatePacket[3];             // to identify multiple identical packets
```

Comment:
rename prevDuplicatePacket to isPrevPacketDuplicate could be preferable because it clearly indicates that the variable represents a boolean status and immediately conveys the purpose of the variable

Reply:
Maybe `prevPacketWasDuplicate`?

Reply:
For booleans it is "programming convention" to prefix it with "is". It is a prefix to see it right away so that it is not hidden in the name. But why you want to use the past term? Is it not a duplicate anymore?

Reply:
@KarateBrot : This for detecting packet that is duplicated more than once. So it is duplicate status of previous iteration.

Reply:
Yeah but that means it is still a duplicate. In my understanding, a duplicated sample will always be a duplicate, no matter how far gone in time it is. For me, "was" implies that it was one thing and then at some point changed to something else. But it is semantics, I guess.

Reply:
Would suggest `isDuplicatePacket`

Did not want to mention `b` as prefix and using CamelCase for the remainder :)
https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig

Reply:
Yes, that boolean means, 'the packet previous to the one we just received `[N-1]` was a duplicate of the one before that `[N-2]`.
In other words, with reference to the current packet, the previous packet was a duplicate (of the one before that).
And it's in the past, not the present, and we cannot change packet contents, so the past tense of 'is' is 'was'.
We just have to abbreviate `thePreviousPacketWasADuplicateOfTheOneBeforeThat` but start the abbreviation with 'is'  lol :-)

Reply:
In the context of the code, perhaps `prevPacketWasADuplicate` seems clear... so we get syntax like this:
```
if (prevPacketWasADuplicate) {
...
```

Reply:
I've implemented that last suggestion, see how it 'reads', I can change to something else if there's a better suggestion.

Reply:
`isPrevPacketDuplicate` should capture `(n-1) == (n-2)`, starts with `is` to identify boolean and reads as "is previous packet duplicate".

Reply:
Yes, but to me, the code is easier to read when it says:
`if prevPacketWasADuplicate` vs `if isPrevPacketDuplicate`
So I'd like to keep it that way, because it's completely easy to read.
You're welcome to make a separate PR to change it, later, but let's go with it like this please  ;-)